lib/decidim/backporter.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+module Decidim
+  class Backporter
+    def initialize(token:, pull_request_id:, version_number:, exit_with_unstaged_changes:)
+      @token = token
+      @pull_request_id = pull_request_id
+      @version_number = version_number
+      @exit_with_unstaged_changes = exit_with_unstaged_changes
+    end
+
+    def call
+      metadata = pull_request_metadata
```

Comment:
Here we would also have to add some check whether the metadata was returned or not:
```suggestion
      metadata = pull_request_metadata
      raise InvalidMetadataError unless metadata

```

Also this would need to be defined:
```ruby
    class InvalidMetadataError < StandardError; end
```